File: javascript/core/src/engine/verbs/concat.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { TableStore } from '../../index.js'
import { SetOp, Step, TableContainer } from '../../types.js'
import { set } from '../util/index.js'

/**
 * Executes an arquero table concat.
 * @param step
 * @param store
 * @returns
 */
export async function concat(
	step: Step,
	store: TableStore,
): Promise<TableContainer> {
	return set(step, store, SetOp.Concat)
}

File: python/data-wrangling-components/data_wrangling_components/engine/verbs/concat.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

import pandas as pd

from data_wrangling_components.engine.verbs.verb_input import VerbInput
from data_wrangling_components.table_store import TableContainer


def concat(input: VerbInput):
    input_table = input.get_input()
    others = input.get_others()
    output = pd.concat([input_table] + others, ignore_index=True)
    return TableContainer(table=output)
